mir/src/node.rs

```diff
@@ -227,16 +228,86 @@ impl MirNode {
                     .1
                     .expect("must have an absolute column ID on base"),
             },
-            MirNodeType::Reuse { ref node } => node.borrow().column_id_for_column(c),
+            MirNodeType::Reuse { ref node } => node.borrow().column_id_for_column(c, table_mapping.clone()),
             // otherwise, just look up in the column set
-            _ => match self.columns.iter().position(|cc| cc == c) {
+            _ =>
+            match self.columns.iter().position(|cc| cc == c) {
                 None => {
-                    panic!(
-                        "tried to look up non-existent column {:?} on node \"{}\" (columns: {:?})",
-                        c, self.name, self.columns
-                    );
-                }
-                Some(id) => id,
+                    // See if table mapping was passed in
+                    match table_mapping {
+                        // if mapping was passed in, then see if c has an associated table, and check
+                        // the mapping for a key based on this
+                        Some(map) => {
+                            match c.table.clone() {
+                                Some(table) => {
+                                    let keyed = format!("{}:{}", c.clone().name, table);
+                                    match map.get(&keyed) {
+                                        Some(ref mut t_name) => {
+
+                                            let mut ind = 0;
+                                            for self_column in &self.columns {
+                                                if self_column.name == c.name && self_column.table == Some(t_name.clone()) {
+                                                    return ind;
+                                                }
+
+                                                ind = ind + 1;
+                                            }
```

Comment:
You can replace this loop by 
```rust
let mut ac = c.clone();
ac.table = Some(t_name.to_owned());
let ind = self.columns.position(|cc| cc == ac).expect("tried to ...");
```
or something similar, if I'm not mistaken, as the loop only computes the index for a matching column.